substrate-archive/src/actors.rs

```diff
@@ -198,13 +199,19 @@ where
 		let listener = Self::init_listeners(conf.pg_url()).await?;
 		let mut conn = pool.acquire().await?;
 		Self::restore_missing_storage(&mut *conn).await?;
-		let env = Environment::<B, R, C, D>::new(conf.backend().clone(), client, actors.storage.clone());
+		let env = Environment::<B, R, C, D>::new(
+			conf.backend().clone(),
+			client,
+			actors.storage.clone(),
+			conf.tracing_targets.clone(),
+		);
 		let env = AssertUnwindSafe(env);
 
 		let runner = coil::Runner::builder(env, crate::TaskExecutor, &pool)
 			.register_job::<crate::tasks::execute_block::Job<B, R, C, D>>()
-			.timeout(std::time::Duration::from_secs(30))
 			.num_threads(conf.workers)
+			.timeout(Duration::from_secs(20))
```

Comment:
Not necessarily everything -- it is a timeout for the blocks execution up until it's sent to the actor that inserts it into the database but it doesn't wait for the insert. It will try to wait for all 64 tasks to _begin_ before timing out. So, it doesn't necessarily wait until all the tasks are finished either, just that they've started to execute on the threadpool

Reply:
Let's add a comment to that effect. :)